File: daily_python/1749_Maximum_Absolute_Sum_of_Any_Subarray.py
import os
import sys
from typing import List

class Solution:
    def maxAbsoluteSum(self, nums: List[int]) -> int:
        min_acc = 0
        max_acc = 0
        prefix = 0
        ans = 0
        for num in nums:
            prefix += num
            max_acc = max(max_acc, prefix)
            min_acc = min(min_acc, prefix)
        return max_acc - min_acc
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            dataset.append(nums)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums in dataset:
        results.append(solution.maxAbsoluteSum(nums))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')

File: daily/2025/1749_Maximum_Absolute_Sum_of_Any_Subarray.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int maxAbsoluteSum(vector<int>& nums) {
        int curr_pos = 0;
        int curr_neg = 0;
        int ans = 0;
        for (int num : nums){
            curr_pos += num;
            curr_neg += num;
            curr_pos = max(curr_pos, 0);
            curr_neg = min(curr_neg, 0);
            ans = max(ans, max(curr_pos, abs(curr_neg)));
        }
        return ans;
    }
};

int main(){
    vector<int> nums = {1,-3,2,3,-4};
    // vector<int> nums = {2,-5,1,-4,3,-2};
    Solution S;
    cout << S.maxAbsoluteSum(nums) << endl;
    return 0;
}